src/toil/test/src/importExportFileTest.py

```diff
@@ -98,6 +99,33 @@ def testImportSharedFileName(self):
             with toil._jobStore.readSharedFileStream(sharedFileName) as f:
                 self.assertEqual(f.read().decode('utf-8'), 'some data')
 
+    def testImportExportFilePermissions(self):
+        """
+        Ensures that uploaded files preserve their file permissions when they
+        are downloaded again. This function checks that an imported executable file
+        maintains its executability after being exported.
+        """
+        options = Job.Runner.getDefaultOptions(self._getTestJobStorePath())
+        options.logLevel = "INFO"
+
+        with Toil(options) as toil:
+            srcFile = '%s/%s%s' % (self._tempDir, 'in', str(uuid.uuid4()))
+            with open(srcFile, 'w') as f:
+                f.write('Hello')
+
+            # Current file permission bits
+            permissionBits = os.stat(srcFile).st_mode
+
+            # Add file owner's execute permissions
+            os.chmod(srcFile, permissionBits | stat.S_IXUSR)
+
+            fileID = toil.importFile('file://' + srcFile)
+            toil.exportFile(fileID, 'file://' + self.dstFile)
+
+            permissionBits = os.stat(self.dstFile).st_mode
+            # Nonzero value indicates executable flag is preserved
+            assert permissionBits & stat.S_IXUSR != 0
```

Comment:
You might want to try this test with a non-exacutable file as well, to make sure that it isn't executable when it comes out.